src/settings_gui.cpp

```diff
@@ -140,6 +142,46 @@ void ShowBaseSetTextfileWindow(TextfileType file_type, const TBaseSet* baseset,
 	new BaseSetTextfileWindow<TBaseSet>(file_type, baseset, content_type);
 }
 
+std::vector<int> _refresh_rates = { 30, 60, 75, 90, 100, 120, 144, 240 };
+
+/* We want 60Hz as the default for this pointer, though it gets overriden with whatever is in the config. */
+int *_current_refresh_rate = &_refresh_rates[1];
+
+static const std::string HERTZ = "Hz";
+std::string _current_refresh_rate_str = std::to_string(*_current_refresh_rate) + HERTZ;
+
+/**
+ * Add the refresh rate from the config and the refresh rates from all the monitors to
+ * our list of refresh rates shown in the GUI.
+ */
+void AddRefreshRatesAndSelect()
+{
+	/* Using a set to make sure no duplicated refresh rates are in the list. */
+	std::unordered_set<int> tempRates(_refresh_rates.begin(), _refresh_rates.end());
+
+	/* Add the refresh rate as selected in the config. */
+	int wantedRate = _settings_client.gui.refresh_rate;
+	tempRates.insert(_settings_client.gui.refresh_rate);
+
+	/* Add all the refresh rates of all monitors connected to the machine.  */
+	std::vector<int> monitorRates = VideoDriver::GetInstance()->GetListOfMonitorRefreshRates();
+	std::copy(monitorRates.begin(), monitorRates.end(), std::inserter(tempRates, tempRates.end()));
+
+	_refresh_rates.clear();
+	_refresh_rates.insert(_refresh_rates.end(), tempRates.begin(), tempRates.end());
```

Comment:
Why take a copy of refresh_rates at all? Why not just add refresh_rate setting & monitorRates to _refreshRates directly, rather than taking a copy of it, clearing it, then readding it
(regardless, this would be better as `_refresh_rate = tempRates;` )

Reply:
Because that would bypass the protection of doubled values of the unordered_set. However, I could really just make _refresh_rates a `std::unordered_set<int>`.

Reply:
Yes, that would probably be better (and what I was assuming was already the case)